### ISSUE TITLE
Updated Document Triggers Inactive

### ISSUE THREAD

**Issue by FlyRocketDrones** [SupremeTechnopriest/react-idle-timer#110]
Hi,

I have a page within my app that shows reports. Reports part of the page gets loaded couple seconds after the page load. Therefore, when IdleTimer has document as element selected, iframe content doesn't exist. Therefore, when Iframe content loads and user is moving their mouse around, IdleTimer does not recognize that as action. Unless user's mouse go outside the iframe area into the document area that existed at initial load. I tried:

 ` const [currentDocument, setcurrentDocument] = useState(document);`

using this to set initial document then update it 1 minute after page initial load by

```
 setTimeout(function () {
    setcurrentDocument(document);
    console.log(123);
    console.log(currentDocument);
  }, 1000 * ); // every 1 minute check
```

But it didn't help either. How can I make sure if document updates IdleTimer grabs the new document?

Above issue opened earlier at https://github.com/SupremeTechnopriest/react-idle-timer/issues/109
I have a follow up questions, therefore opening up another issue tracker.

Iframe content is coming from a server hosted Analytics platform. I am not able to mount IdleTimer 
to the platform as you suggested. Why IdleTimer cannot monitor if document changed? 

So IdleTimer will not work if DOM is updated after load time. Once I give reference of Document to the IdleTimer if I change the Document, add a new DIV element per say, IdleTimer will not recongize movements within that new DIV. That is a huge bug. IdleTimer needs to get current document and possibly check every lets say 1 minute to see if Document updated and if so get the new Document as a reference with all it's child elements, at a basic level I am thinking. There might be so many other ways to handle this issue. So what you'll suggest, nobody can add new elements to their DOM because IdleTimer cannot track movements on those new items? That's a major issue. If this issue gets closed like the one in the link mentioned, everyone should note, developer doesn't care to correct bugs and be aware of the limitation I mentioned above before using IdleTimer. I am moving my suite of apps away until this issue fixed.

**Comment by SupremeTechnopriest** [comment]
Hey hey!  You can still comment on closed issues, but we will handle it here. I think I see what you are asking for.  Can you try to call:
```js
idleTimer._unbindEvents()
idleTimer._bindEvents()
```

after you change the element?

If that works for you I can push a change that will allow you to change the element mid lifecycle.  Currently, events are only bound when the component mounts and unbound when it unmounts. 

See here: https://github.com/SupremeTechnopriest/react-idle-timer/blob/master/src/index.js#L227-L235

and here: https://github.com/SupremeTechnopriest/react-idle-timer/blob/master/src/index.js#L227-L235

A likely fix would be changing this: https://github.com/SupremeTechnopriest/react-idle-timer/blob/master/src/index.js#L237-L246

to this:

```js
componentDidUpdate (prevProps) {
    // Update debounce function
    if (prevProps.debounce !== this.props.debounce) {
      this.debouncedAction = debounced(this.props.onAction, this.props.debounce)
    }
    // Update throttle function
    if (prevProps.throttle !== this.props.throttle) {
      this.throttledAction = throttled(this.props.onAction, this.props.throttle)
    }
    if (prevProps.element !== this.props.element) {
        this.componentWillUnmount()
        this.componentDidMount()
    }
  }
```

**Comment by SupremeTechnopriest** [comment]
@FlyRocketDrones Any luck?

**Comment by SupremeTechnopriest** [comment]
@FlyRocketDrones I guess you worked out your issue?  You can reopen this issue if you still need this feature.

**Comment by FlyRocketDrones** [comment]
@SupremeTechnopriest I did. The problem was that, we have below code in the AuthService
```
    if (
      localStorage.getItem("msal.error.description") != null &&
      localStorage.getItem("msal.error.description").indexOf("AADB2C90118") > -1
    ) {
      this.passwordReset();
```

Once user enters in Password reset they continue to have AADB2C90118 errorcode even after completion of the password reset. So we added redirectUri to the authconfig of the msal and now user get's redirected to www.website.com/PasswordReset and added PasswordReset as a route to run below code

```
    localStorage.clear();
    this.clientApplication.logout(); 
```

Now user gets a screen to log back in.
